### PR TITLE
SCTPTransport: add getAssociation

### PR DESCRIPTION
**Pull request by zyxar**
#### Description
Use assocation() instead of accessing private struct member

member variable `association` in SCTPTransport renamed to `sctpAssociation`

This is a follow-up commit to #1747

### CODE REVIEW

**Comment by zyxar** [comment]
or is it better to rename `getAssocation()` as `Assocation()`?

**Comment by Sean-Der** [comment]
@zyxar that works for me! I would leave it private, I believe Go will have a conflict then though method+member named `assocation`

File: peerconnection_media_test.go [hunk 1]
```diff
@@ -339,7 +339,7 @@ func TestPeerConnection_Media_Disconnected(t *testing.T) {
 			for {
 				if len(vp8Sender.Transport().GetRemoteCertificate()) != 0 {
 					pcAnswer.sctpTransport.lock.RLock()
```
Comment (Sean-Der) [inline peerconnection_media_test.go:341]: Do we still need this lock?

File: sctptransport.go [hunk 2]
```diff
@@ -369,3 +366,13 @@ func (r *SCTPTransport) generateAndSetDataChannelID(dtlsRole DTLSRole, idOut **u
 
 	return &rtcerr.OperationError{Err: ErrMaxDataChannelID}
 }
+
+func (r *SCTPTransport) Association() *sctp.Association {
```
Comment (Sean-Der) [inline sctptransport.go:370]: I don't believe we should export this. It isn't part of the ORTC ond webrtc-pc API

File: peerconnection_media_test.go (hunk 1 above)
Comment (zyxar) [reply peerconnection_media_test.go]: ah, no longer needed.

File: sctptransport.go (hunk 2 above)
Comment (zyxar) [reply sctptransport.go]: 👌
renamed to unexported `association()`

**Review by Sean-Der — APPROVED** [review]

**Comment by Sean-Der** [comment]
:fire: 

Love these improvements. Feel free to file any improvements you think would be good. Even if they are big projects we can take them on together.

**Pull request merged**